### ISSUE TITLE
unix ref count woes

### ISSUE THREAD

**Issue by ry** [joyent/libuv#45]
Our current strategy has been `uv_ref` on each handle that is initialized. In libev watchers also increase the ref count when they are active. Should a handle increase the count by only one ever? Or is it allowed to increase it more? Since the unix backend is currently increasing the count more than once, Node does not exit naturally.

We need tests to verify we're acting the same on both platforms. Something like this:

```
TEST_IMPL(ref) {
  uv_init(alloc_cb);
  uv_idle_init(&idle_2_handle, NULL, NULL);
  uv_run();
  return 0;
}


TEST_IMPL(ref2) {
  uv_init(alloc_cb);

  uv_idle_init(&idle_2_handle, NULL, NULL);
  uv_idle_start(&idle_2_handle, NULL);
  uv_unref(&idle_2_handle);

  uv_run();
  return 0;
}
```

Note ref2 currently times out on unix.

**Comment by piscisaureus** [comment]
I would suggest to increase the handle count _only_ when a handle is initialized, then decrease it only after it's close callback is called. Starting / stopping should not affect the counter. Let me know if that is doable on the unix side.

**Comment by ry** [comment]
Would you pass the above test currently?

Yes - that is doable.

**Comment by piscisaureus** [comment]
ref2 would be ok but ref (1) would time out, because initializing a handle adds a ref, and that handle is never closed. (Whether the idle watcher is active doesn't matter.)

**Comment by ry** [comment]
fixed in 40f0ad6d364cff76adaf1c7f182b1dc939f36617
